pkg/crc/segment/segment.go

```diff
@@ -139,6 +139,10 @@ func readIdentifyHash(identifyHashPath string) (uint64, error) {
 		return 0, err
 	}
 
+	if len(cachedHashBytes) != 8 {
+		return 0, errors.New("Got unexpected cached hash bytes")
```

Comment:
`fmt.Errorf("Got %d bytes hash, expected 8", len)` maybe?

Reply:
Reading more closely segment.go's code, we should return `0, nil` when we read 0 bytes, this will allow the telemetry code to generate a new hash when it needs to upload instead of erroring out.
Maybe we should always ignore errors from readIdentifyHash so that telemetry works/we try to generate a new hash?

Reply:
Errors here are bad, as it prevents us from collecting them

So hardening teh code in general is important. We seemed to have recorded an empty value which should not have happened to begin with.

Note, This happened for me also on Windows.. and on an existing installation (which had a hash).